### ISSUE TITLE
docker-rsync always crashes after initial sync?

### ISSUE THREAD

**Issue by shakefu** [synack/docker-rsync#16]
Here's the output I get:

```
$ docker-rsync -verbose -dst /mnt/sda1/rsync/ dev
docker-machine ssh dev 'tce-load -wi rsync attr acl'
docker-machine ssh dev 'sudo mkdir -p /mnt/sda1/rsync'
Syncing /Users/SNIP (local) to /mnt/sda1/rsync/ (docker-machine dev)
Watching for file changes ...
panic: runtime error: cgo argument has Go pointer to Go pointer

goroutine 1 [running]:
panic(0x413c2c0, 0xc8200148c0)
	/usr/local/Cellar/go/1.6/libexec/src/runtime/panic.go:464 +0x3e6
github.com/mattes/fsevents.(*EventStream).Start(0xc82000e3c0)
	/private/tmp/docker-rsync20160221-26000-14w77a5/docker-rsync-0.1.1/Godeps/_workspace/src/github.com/mattes/fsevents/fsevents.go:208 +0x529
main.Watch(0xc8200100c4, 0x28, 0xc820016a40)
	/private/tmp/docker-rsync20160221-26000-14w77a5/docker-rsync-0.1.1/watch.go:41 +0x170
main.main()
	/private/tmp/docker-rsync20160221-26000-14w77a5/docker-rsync-0.1.1/main.go:93 +0x14aa
```

Edit: Should add that the initial sync worked, files are on the VM, but it won't watch for changes.

**Comment by shakefu** [comment]
Apparently this is happening in fsnotify/fsevents#12 as well, due to Go 1.6.

**Comment by stkiss** [comment]
Seeing the same issue. This bug makes the entire project unusable.

**Comment by mattes** [comment]
Waiting for https://github.com/fsnotify/fsevents/pull/13 to be done. In the meanwhile, please don't use Go 1.6.

**Comment by HenrichN** [comment]
Same issue here. Luckily we had an older version lying around build with Go 1.5...

**Comment by wadey** [comment]
@mattes:

> Waiting for fsnotify/fsevents#13 to be done. In the meanwhile, please don't use Go 1.6.

That fix has been merged.

**Comment by 3den** [comment]
Any update on this?

**Comment by wadey** [comment]
Someone needs to change this to `github.com/fsnotify/fsevents`: https://github.com/synack/docker-rsync/blob/c01d8a45be8d4bef3dd1d976aca23db1fd3b85b2/Godeps/Godeps.json#L6-L7 and update the SHA

**Comment by wadey** [comment]
nm it looks like @mattes needs to update his forked with the fix, as this repo depends on his fork.

**Comment by 3den** [comment]
seems like no one is maintaining this project so if some of you guys fork it that would be awesome

**Comment by ishristov** [comment]
Anyone else still seeing this? Not changed for me..

```
Watching for file changes ...
                                                 panic: runtime error: cgo argument has Go pointer to Go pointer

                                                                                                                goroutine 1 [running]:
              panic(0x413c2c0, 0xc82009a290)
                                            	/usr/local/Cellar/go/1.6/libexec/src/runtime/panic.go:464 +0x3e6
                                                                                                                github.com/mattes/fsevents.(*EventStream).Start(0xc8200ac0c0)
                                                     	/private/tmp/docker-rsync20160221-28571-qmpmy6/docker-rsync-0.1.1/Godeps/_workspace/src/github.com/mattes/fsevents/fsevents.go:208 +0x529
                                                                         main.Watch(0x7fff5fbffb6c, 0x3, 0xc82009c380)
                                                                                                                      	/private/tmp/docker-rsync20160221-28571-qmpmy6/docker-rsync-0.1.1/watch.go:41 +0x170
                                                                                   main.main()
                                                                                              	/private/tmp/docker-rsync20160221-28571-qmpmy6/docker-rsync-0.1.1/main.go:93 +0x14aa

```

**Comment by 3den** [comment]
@ishristov @wadey this project is not required, there is a much easier way to sync docker with your local directory https://gist.github.com/3den/2dbebfa2be0846c51a737d595ad362c4

**Comment by aterreno** [comment]
same

**Comment by mattes** [comment]
fixed in #19
